### PR TITLE
Input validation for Version strings

### PR DESCRIPTION
**Pull request by chrisohaver**

### CODE REVIEW

File: kubernetes/migration/migrate.go [hunk 1]
```diff
@@ -229,3 +239,34 @@ func Released(dockerImageID string) bool {
 	}
 	return false
 }
+
+// ValidVersions returns a list of all versions defined
+func ValidVersions() []string {
```
Comment (rajansandeep) [inline kubernetes/migration/migrate.go]: This function isn't used currently. 
Will this be used for special cases such as v1.2.3?

File: kubernetes/migration/migrate_test.go [hunk 2]
```diff
@@ -253,3 +252,27 @@ stubzone.org:53 {
...
+func TestValidateVersions(t *testing.T) {
+	testCases := []struct {
+		from   string
+		to     string
+		shouldErr    bool
+	}{
+		{"1.3.1", "1.5.0", false},
+		{"banana", "1.5.0", true},
+		{"1.3.1", "apple", true},
+		{"banana", "apple", true},
```
Comment (rajansandeep) [inline kubernetes/migration/migrate_test.go]: Maybe a couple more test cases where coredns to version <= coredns from version (1.4.0->1.3.1)

File: kubernetes/migration/migrate_test.go [hunk 3]
```diff
@@ -253,3 +252,27 @@ stubzone.org:53 {
...
+	}
+
+	for _, tc := range testCases {
+		err := validateVersions(tc.from, tc.to)
+
+		if !tc.shouldErr && err != nil {
+			t.Errorf("expected to '%v' to '%v' to be valid versions.", tc.from, tc.to)
```
Comment (rajansandeep) [inline kubernetes/migration/migrate_test.go]:
```suggestion
			t.Errorf("expected '%v' to '%v' to be valid versions.", tc.from, tc.to)
```

File: kubernetes/migration/migrate_test.go [hunk 4]
```diff
@@ -253,3 +252,27 @@ stubzone.org:53 {
...
+		}
+		if tc.shouldErr && err == nil {
+			t.Errorf("expected to '%v' to '%v' to be invalid versions.", tc.from, tc.to)
```
Comment (rajansandeep) [inline kubernetes/migration/migrate_test.go]:
```suggestion
			t.Errorf("expected '%v' to '%v' to be invalid versions.", tc.from, tc.to)
```

File: kubernetes/migration/migrate.go (hunk 1 above)
Comment (chrisohaver) [reply kubernetes/migration/migrate.go]: Yes. 1.2.3 is a valid version - it's a valid _start_ version.  This just returns a simple list of versions that the tool supports, for example, a CLI tool.  It doesn't convey which combinations of versions are valid together.

**Comment by chrisohaver** [comment]
had to cleanup my master. Closing this PR and re-opening in another branch with clean master history.

**Pull request closed without merging**